scala/scala.bzl

```diff
@@ -554,6 +554,40 @@ def _scala_test_impl(ctx):
     )
     return _scala_binary_common(ctx, cjars, rjars)
 
+def _gen_test_suite_flags_based_on_prefixes_and_suffixes(ctx, archive):
+    return struct(suite_class = "io.bazel.rulesscala.test_discovery.DiscoveredTestSuite", 
+    archiveFlag = "-Dbazel.discover.classes.archive.file.path=%s" % archive.short_path,
+    prefixesFlag = "-Dbazel.discover.classes.prefixes=%s" % ",".join(ctx.attr.prefixes),
+    suffixesFlag = "-Dbazel.discover.classes.suffixes=%s" % ",".join(ctx.attr.suffixes),
+    printFlag = "-Dbazel.discover.classes.print.discovered=%s" % ctx.attr.print_discovered_classes)
+
+def _scala_junit_test_impl(ctx):
+    if (not(ctx.attr.prefixes) and not(ctx.attr.suffixes)):
+      fail("Setting at least one of the attributes ('prefixes','suffixes') is required")
+    deps = ctx.attr.deps + [ctx.attr._suite]
+    jars = _collect_jars(deps)
+    (cjars, rjars) = (jars.compiletime, jars.runtime)
+    junit_deps = [ctx.file._junit,ctx.file._hamcrest]
+    cjars += junit_deps
+    rjars += [ctx.outputs.jar,
+              ctx.file._scalalib
+              ] + junit_deps
+    rjars += _collect_jars(ctx.attr.runtime_deps).runtime
+    test_suite = _gen_test_suite_flags_based_on_prefixes_and_suffixes(ctx, ctx.outputs.jar)
+    launcherJvmFlags = ["-ea", test_suit.archiveFlag, test_suite.prefixesFlag, test_suite.suffixesFlag, test_suite.printFlag]
```

Comment:
Thanks!
Used github resolve conflicts and just copy pasted your snippet :( I checked it but missed that. Resolved.